extensions/applicationinsights-clickanalytics-js/src/Interfaces/Datamodel.ts

```diff
@@ -0,0 +1,382 @@
+    cN?: string;
+    /**
+     * Unique identifier (friendly names allowed) of the content (Content Id) as designated by the content provider. 
+     * The name should be unique per contentSource such that the two make up the compound key and such that the contentId can be 
+     * looked up in the contentSource for additional metadata.
+     */
+    id?: string;
+    /**
+     * User specified custom content properties
+     */
+    [name: string]: string | number | boolean | string[] | number[] | boolean[] | object;
+}
+
+ /**
+  * Content handler interface
+  */
+export interface IContentHandler {
+    /**
+     * Get meta data 
+     */
+    getMetadata: () => { [name: string]: string };
+    /**
+     * Get element content 
+     */
+    getElementContent: (element: Element, eventType?: EventType) => IContent;
+    
+  }
+
+/**
+ * Page Action event
+ */
+export interface IPageActionTelemetry extends ITelemetryEventInternal {
+    /**
+     * Target uri for PageAction events
+     */
+    targetUri?: string;
+    /**
+     *  One of the awa.actionType values
+     */
+    actionType?: string;
+    /**
+     * One of the awa.behavior values. 
+     */
+    behavior?: number;
+    /**
+     * X, Y representing the absolute co-ordinates withrespect to topleft corner of the page. This should be normalized for the screen resolution to provide better heat map.
+     */
+    clickCoordinates?: string;
+    /**
+     * JSON-formatted array of content acted upon 
+     */
+    content?: any;
+    /**
+     * Version indicating content version which aids in parsing the content.
+     */
+    contentVer?: string;
+    /**
+     * Uri of the referrer, this is a convinence for adaptors to just leverage PageAction for click analytics without linking it to the respective Page Views.
+     */
+    refUri?: string;
+    /**
+     * Time taken in milliseconds since the user saw the page and took the action (such as click on a link, etc.). This will be in seconds.
+     */
+    timeToAction?: number;
+}
+
+export interface ITelemetryEventProperties {
+    /**
+     * User specified custom properties
+     */
+    [name: string]: string | number | boolean | string[] | number[] | boolean[] | object;
+}
+
+/**
+ * Page Action event properties (part C)
+ */
+export interface IPageActionProperties extends ITelemetryEventProperties {
+    /**
+     * Uri of the referrer, this is a convinence for adaptors to just leverage PageAction for click analytics without linking it to the respective Page Views.
+     */
+    refUri?: string;
+    /**
+     * Time taken in milliseconds since the user saw the page and took the action (such as click on a link, etc.). This will be in seconds.
+     */
+    timeToAction?: number;
+    /**
+     * Server impression Guid, as set by the adopter
+     */
+    serverImpressionGuid?: string;
+    /**
+     * List of cookies in semi-colon delimited name value pair
+     */
+    cookies?: string;
+}
+
+export interface ITelemetryEvent {
```

Comment:
Same here, I would expect this one to be in core already, no need to redefine this interface